docs/encyclopedia/workflow/workflow-execution/event.mdx

```diff
@@ -90,19 +92,11 @@ Events are created by the Temporal Service in response to either Commands or act
 
 **Is there a limit to how long Workflows can run?**
 
-No, there is no time constraint on how long a Workflow Execution can be Running.
-
-However, Workflow Executions intended to run indefinitely should be written with some care.
-The Temporal Service stores the complete Event History for the entire lifecycle of a Workflow Execution.
-The Temporal Service logs a warning after 10Ki (10,240) Events and periodically logs additional warnings as new Events are added.
-If the Event History exceeds 50Ki (51,200) Events, the Workflow Execution is terminated.
+No, there is no time constraint on how long a Workflow Execution can run.
 
-To prevent _runaway_ Workflow Executions, you can use the Workflow Execution Timeout, the Workflow Run Timeout, or both.
-A Workflow Execution Timeout can be used to limit the duration of Workflow Execution Chain, and a Workflow Run Timeout can be used to limit the duration an individual Workflow Execution (Run).
+However, Workflow Executions intended to run indefinitely should be written with some care as they can run into [Event History limits](#event-history-limits).
```

Comment:
Could make this a bit more concrete:

```suggestion
However, if your Workflow will perform many actions, or will receive many messages, then you should use Continue-As-New to avoid hitting [Event History limits](#event-history-limits) and performance degredation.
```